### PR TITLE
che #92 Adding Keycloak exception. Refactoring

### PR DESCRIPTION
**Pull request by ibuziuk**

### CODE REVIEW

**Comment by ibuziuk** [comment]
@mlabuda  `StackControllerTest.testGetStacksWithWrongToken` start failing for me :/

**Comment by mlabuda** [comment]
It is caused because if I get wrong token, I expected 401 - Unauthorized HTTP status code. In this PR you changed it to internal server HTTP status code, what is different. We should have an agreement how to handle specific type of error and what status code we should return.

File: src/main/java/io/fabric8/che/starter/exception/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -75,4 +75,10 @@ public String handleProjectCreationException(ProjectCreationException e) {
         return e.getMessage();
     }
 
+    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR, reason = "Invalid response from Keycloak server")
```
Comment (mlabuda) [inline src/main/java/io/fabric8/che/starter/exception/GlobalExceptionHandler.java]: Maybe we should return unauthorized error, if response from Keycloak does no return token. Also reason is not valid here - response was not invalid, response was correct, but the token passed to keycloak was invalid and thus the response was `{"errorMessage":"Invalid token."}`

**Comment by ibuziuk** [comment]
@mlabuda so, if the invalid token is provided to Keycloak server response body would be the `{"errorMessage":"Invalid token."}` (or is it only the behavior of the test server that emulates Keycloak one)?

**Comment by mlabuda** [comment]
I tried real keycloak server and that is the response. Test server just simulates the real one with the same behavior.

**Comment by ibuziuk** [comment]
@mlabuda PR is updated - could you take another look ?

**Pull request merged**